packages/sdk/src/types/runtime/assets/assetManager.ts

```diff
@@ -94,6 +97,71 @@ export class AssetManager {
         return createForwardPromise(asset, promise);
     }
 
+    public loadSoundAsset(groupName: string, uri: string, options: LoadSoundOptions): ForwardPromise<string> {
+
+        log.error('app', "STARTING LOADSOUNDASSET");
+        const soundAssetId = UUID();
+        this.enqueueLoadSoundPromise(
+            soundAssetId, {
+                resolve: () => { /* empty */ },
+                reject: () => { /* empty */ },
+            });
+        const promise = this.sendLoadAssetsPayload({
+            type: 'load-sound',
+            id: soundAssetId,
+            uri
+        } as LoadSound)
+            .then<string>(payload => {
+                if (payload.failureMessage || payload.assets.length !== 1) {
+                    this.notifySoundLoaded(soundAssetId, false, payload.failureMessage);
+                    return Promise.reject(`Loading sound asset ${uri} failed: ${payload.failureMessage}`);
+                }
+                // soundAssetId = payload.assets[0].id;
+                this.notifySoundLoaded(soundAssetId, true);
+            });
+        this.registerLoadPromise(promise);
+
+        return createForwardPromise(soundAssetId, promise);
+    }
+
+    public unloadSoundAsset(handle: string) {
+    }
+
+    public loadSoundPromises: { [name: string]: ExportedPromise[] };
+
+    public notifySoundLoaded(soundAssetId: string, success: boolean, reason?: any): void {
+        if (!!this.loadSoundPromises && !!this.loadSoundPromises[soundAssetId]) {
+            const loadSoundPromises = this.loadSoundPromises[soundAssetId].splice(0);
+            delete this.loadSoundPromises[soundAssetId];
+            for (const promise of loadSoundPromises) {
+                if (success) {
+                    promise.resolve();
+                } else {
+                    promise.reject(reason);
+                }
+            }
+        }
+    }
+
+    public enqueueLoadSoundPromise(soundAssetId: string, promise: ExportedPromise): void {
+        if (!this.loadSoundPromises) {
+            this.loadSoundPromises = {};
+        }
+        if (!this.loadSoundPromises[soundAssetId]) {
+            this.loadSoundPromises[soundAssetId] = [];
+        }
+        this.loadSoundPromises[soundAssetId].push(promise);
+    }
+
+    public SoundLoaded(soundAssetId: string): Promise<void> {

```

Comment:
>SoundLoaded [](start = 11, length = 11)

A better API design would be for AssetManager to have a single `loaded` method that supports all asset types. Will require some structural changes to AssetManager itself.